src/ark/utils/plot_utils.py

```diff
@@ -147,14 +144,30 @@ def _metacluster_cmap_generator(self) -> None:
             f"{self.cluster_type}_meta_cluster"
         ].map(self.metacluster_colors)
 
-        # Convert the list of tuples to a numpy array, each index is a color
+        # grab mask cluster_id integers and merge with raw_cmap
+        cluster_id_to_metacluster_map = cluster_id_to_name[
+            [f"{self.cluster_type}_meta_cluster", "cluster_id"]].drop_duplicates()
+        unassigned_cluster_id: int = int(cluster_id_to_name["cluster_id"].max() + 1)
+        cluster_id_to_metacluster_map = pd.concat(
+            [
+                cluster_id_to_metacluster_map,
+                pd.DataFrame(
+                    data={
+                        f"{self.cluster_type}_meta_cluster": [unassigned_cluster_id, 0],
+                        "cluster_id": [unassigned_id, 0],
```

Comment:
I think `unassgined_cluster_id` and `unassigned_id` are flipped? Should it be:
```
data={
                        f"{self.cluster_type}_meta_cluster": [unassigned_id, 0],
                        "cluster_id": [unassigned_cluster_id, 0],
}
```

Since above, `unassigned_cluster_id: int = int(cluster_id_to_name["cluster_id"].max() + 1)`, so `unassigned_cluster_id` shoud correspond to the `cluster_id` column.

(but per my comment below, this might not be necessary at all)